### PR TITLE
avm2: Add `flash.utils.[set|clear][Timeout|Interval](...)`

### PR DESCRIPTION
**Pull request by torokati44**
This makes the "MOUSE OVER FOR INFO" text disappear after a short time, as it should, on z0r loops: 7019, 7641, and 7489

One weird thing I noticed, is that the timers somehow start right when the movie is started to get loaded, and they roll even while the "AS3 compatibility warning" popup is up, and blocks the player from displaying anything.

So, with these loops, if you don't dismiss that warning quick enough, or you start Ruffle in debug mode, you might not even see the text overlay at all.

### CODE REVIEW

**Comment by torokati44** [comment]
TODO: Check what should happen if those `unwrap`s were to fail.

**Comment by adrian17** [comment]
My understanding is that currently we don't do parameter type checks on native functions, so yeah, you should do `as_function_object` on the first argument and `coerce_to_number` on second, with error propagation everywhere.

**Comment by torokati44** [comment]
Okay, thanks for the tip! So, just like this?

File: core/src/avm2/globals/flash/utils.rs [hunk 1]
```diff
@@ -25,6 +25,66 @@ pub fn get_timer<'gc>(
         .into())
 }
 
+/// Implements `flash.utils.setInterval`
+pub fn set_interval<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    let (args, params) = args.split_at(2);
```
Comment (EmperorBale) [inline core/src/avm2/globals/flash/utils.rs:34]: This panics if there are less than 2 arguments, lets check the length and return an error if there are less than 2 before doing this

File: core/src/avm2/globals/flash/utils.rs [hunk 2]
```diff
@@ -25,6 +25,66 @@ pub fn get_timer<'gc>(
...
+    )))
+}
+
+/// Implements `flash.utils.clearInterval`
+pub fn clear_interval<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    let id = args.get(0).unwrap().coerce_to_number(activation)?;
```
Comment (EmperorBale) [inline core/src/avm2/globals/flash/utils.rs:53]: Should make sure we have an argument beforehand, or do handle not having an argument properly.

File: core/src/avm2/globals/flash/utils.rs [hunk 3]
```diff
@@ -25,6 +25,66 @@ pub fn get_timer<'gc>(
...
+    Ok(Value::Undefined)
+}
+
+/// Implements `flash.utils.setTimeout`
+pub fn set_timeout<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    let (args, params) = args.split_at(2);
```
Comment (EmperorBale) [inline core/src/avm2/globals/flash/utils.rs:64]: (same as before)

File: core/src/avm2/globals/flash/utils.rs [hunk 4]
```diff
@@ -25,6 +25,66 @@ pub fn get_timer<'gc>(
...
+    )))
+}
+
+/// Implements `flash.utils.clearTimeout`
+pub fn clear_timeout<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    let id = args.get(0).unwrap().coerce_to_number(activation)?;
```
Comment (EmperorBale) [inline core/src/avm2/globals/flash/utils.rs:83]: (same as before)

**Comment by torokati44** [comment]
Thank you for the review @EmperorBale! Does this look okay now?

**Review by EmperorBale — APPROVED** [review]
LGTM

**Comment by Herschel** [comment]
Thank you!

**Pull request merged**